### PR TITLE
feat: improve modulo for chain

### PR DESCRIPTION
**Pull request by raizo07**
Time spent on this PR:

Resolves: #[1408](https://github.com/kkrt-labs/kakarot-rpc/issues/1408)

# Pull Request type

Please check the type of change your PR introduces:

- [x] Feature

# What is the new behavior?

The pr cleans up the modulo logic and enhances the overall code

# Does this introduce a breaking change?

- [x] No

### CODE REVIEW

File: .python-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.10.10
```
Comment (tcoratger) [inline .python-version:1]: Can you remove this file?

File: src/bin/hive_chain.rs [hunk 2]
```diff
@@ -54,6 +54,14 @@ pub struct Args {
 const STARKNET_RPC_URL: &str = "http://0.0.0.0:5050";
 const MAX_FELTS_IN_CALLDATA: &str = "30000";
 
+// Define the modulo constant
+const CHAIN_ID_MODULO: u64 = 1u64 << 53;
```
Comment (tcoratger) [inline src/bin/hive_chain.rs:58]: Should be (1u64 << 53) - 1

File: src/bin/hive_chain.rs [hunk 3]
```diff
@@ -54,6 +54,14 @@ pub struct Args {
...
+
+// Utility function to apply the chain ID modulo
+fn apply_chain_id_modulo(chain_id: u64) -> u64 {
```
Comment (greged93) [inline src/bin/hive_chain.rs:61]: Not needed just for one call

File: src/bin/hive_chain.rs [hunk 4]
```diff
@@ -71,6 +79,9 @@ async fn main() -> eyre::Result<()> {
 
     // Set the chain id
     let chain_id = starknet_provider.chain_id().await?;
+    let chain_id_mod = apply_chain_id_modulo(chain_id);
+
+    // Initialize the chain ID globally
     let modulo = (1u64 << 53) - 1;
```
Comment (greged93) [inline src/bin/hive_chain.rs:85]: Remove the lines here since you already have the chain id mod

**Review by greged93 — COMMENTED** [review]
Can you also search for the other place where we use modulo 2^53 - 1 and replace with the constant?

File: src/bin/hive_chain.rs (hunk 4 above)
Comment (raizo07) [reply src/bin/hive_chain.rs:85]: Okay

**Comment by greged93** [comment]
hey sorry @raizo07, this is getting urgent and we need to make more changes than planned, which close your PR and make another one

**Comment by raizo07** [comment]
Hey @greged93 I had already fixed the required changes.

**Comment by greged93** [comment]
right but we needed to add quite a lot and clean a lot of code.

**Comment by raizo07** [comment]
Can I message you on Telegram? What's your handle.

**Pull request closed without merging**